core/node/fee_model/src/l1_gas_price/gas_adjuster/mod.rs

```diff
@@ -18,6 +18,75 @@ mod metrics;
 #[cfg(test)]
 mod tests;
 
+#[derive(Debug)]
+pub enum GasAdjusterClient {
```

Comment:
I need to clarify: when I've said to remove the enum, I've meant simplifying it to

```rust
#[derive(Debug)]
pub struct GasAdjusterClient {
    inner: Box<dyn EthFeeInterface>,
}
```

...which is AFAIU is possible. It has downsides, e.g., it requires double-boxing `Box<DynClient<_>>`, which AFAIU is impossible to avoid because Rust doesn't convert among trait objects, but IMO is overall cleaner. If L1 / L2 tags need to be retained (not 100% sure whether it's possible to refactor assertions to avoid them), they could be stored in an additional field. Alternatively, int may be made a part of `EthFeeInterface` itself (i.e., add `fn network_type(&self) -> L1 | L2`); not sure whether this makes sense semantically. Regarding component tags, I think it's reasonable to have this tagging moved to the caller side.

Reply:
> Regarding component tags, I think it's reasonable to have this tagging moved to the caller side.

It is just not consistent with the code we have right now and more error-prone (easier to remember to tag in one place instead of N places)

Reply:
I agree with @StanislavBreadless

Reply:
Fine, it can still work with `struct GasAdjusterClient` if we don't do `Box<dyn EthFeeInterface>` dependency injection; in this case, the tag will be set when converting an L1 / L2 client to `Box<dyn EthFeeInterface>`.

Reply:
Okay, I put the tag into the conversion methods